### PR TITLE
SDK-1985 Fetching session config support

### PR DESCRIPTION
**Pull request by abdalmajeed-yoti**
Support for fetching session configuration.

### CODE REVIEW

File: src/client/idv.client.js [hunk 1]
```diff
@@ -92,6 +92,18 @@ class IDVClient {
...
     return this.idvService.getSupportedDocuments();
   }
+
+  /**
+   * Fetches the configuration for the given sessionID.
+   *
+   * @param {string} sessionId
+   * @return {SessionConfigurationResponse}
+   *
+   * @returns {Promise}
```
Comment (laurent-yoti) [inline src/client/idv.client.js:102]: I used a bit the `examples/idv` to try it out, works fine!
But I realised that we are not really doing a good job with the JSDoc when it comes to promise.

Could you please update the `idv.client.js` and `idv.service.js` JSDoc comments so:
- there is only 1 `@return`
- use `return` or `returns` (which ever is the one the most used in the project)
- replace any `@returns {Promise}` by `@returns {Promise<TheClassItResolvesWith>}`

I had a look about the reject/error, but there is nothing available in JSDoc it seems... so, we'll get these when on Typescript later.

File: src/idv_service/session/retrieve/configuration/capture/required.resource.response.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+   * Returns if the Relying Business is allowed to upload resources
+   * to satisfy the requirement.
+   *
+   * return the end user is allowed to upload resources
+   *
+   * @return {boolean}
+   */
+  isRelyingBusinessAllowed() {
+    if (this.allowedSources) {
+      return this.allowedSources.findIndex(
```
Comment (laurent-yoti) [inline src/idv_service/session/retrieve/configuration/capture/required.resource.response.js:88]: Could be replace with `.some`

File: tests/idv_service/session/retrieve/configuration/capture/document/objective.response.spec.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const ObjectiveResponse = require('../../../../../../../src/idv_service/session/retrieve/configuration/capture/document/objective.response');
+
+describe('ObjectiveResponse', () => {
+  let objectiveResponse;
+
+  beforeEach(() => {
+    objectiveResponse = new ObjectiveResponse({
+      type: '',
```
Comment (laurent-yoti) [inline tests/idv_service/session/retrieve/configuration/capture/document/objective.response.spec.js:8]: why keeping it as empty string ?

File: tests/idv_service/session/retrieve/configuration/session.configuration.response.spec.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+describe('SessionConfigurationResponse', () => {
+  let sessionConfigurationResponse;
+
+  beforeEach(() => {
+    sessionConfigurationResponse = new SessionConfigurationResponse({
+      client_session_token_ttl: 123,
+      session_id: 'some_id',
+      requested_checks: [
+        '',
```
Comment (laurent-yoti) [inline tests/idv_service/session/retrieve/configuration/session.configuration.response.spec.js:12]: That seems bizarre... should we have a least 1 check, so then we can verify the length and the type of check it was converted to ?

File: tests/idv_service/session/retrieve/configuration/capture/document/required.supplementary.document.resource.response.spec.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  let requiredSupplementaryDocumentResourceResponse;
+
+  beforeEach(() => {
+    // eslint-disable-next-line max-len
+    requiredSupplementaryDocumentResourceResponse = new RequiredSupplementaryDocumentResourceResponse({
+      type: 'SUPPLEMENTARY_DOCUMENT',
+      id: '',
+      state: '',
+      document_types: [
+        '',
```
Comment (laurent-yoti) [inline tests/idv_service/session/retrieve/configuration/capture/document/required.supplementary.document.resource.response.spec.js:13]: Same comment about empty values...

File: tests/idv_service/session/retrieve/configuration/capture/document/required.supplementary.document.resource.response.spec.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      ],
+      country_codes: [
+        'GBR',
+      ],
+      objective: {
+        type: '',
```
Comment (laurent-yoti) [inline tests/idv_service/session/retrieve/configuration/capture/document/required.supplementary.document.resource.response.spec.js:19]: Same comment about empty values...

File: tests/idv_service/session/retrieve/configuration/session.configuration.response.spec.js (hunk 4 above)
Comment (abdalmajeed-yoti) [reply tests/idv_service/session/retrieve/configuration/session.configuration.response.spec.js:12]: There is no conversion for the requested checks.

File: src/idv_service/session/retrieve/configuration/capture/source/allowed.source.response.js [hunk 7]
```diff
@@ -1,4 +1,11 @@
+const Validation = require('../../../../../../yoti_common/validation');
+
 class AllowedSourceResponse {
+  constructor(type) {
+    Validation.isString(type, 'type');
+    this.type = type;
+  }
```
Comment (laurent-yoti) [inline src/idv_service/session/retrieve/configuration/capture/source/allowed.source.response.js:7]: So, that is fine, but this is not quite an Abstract class... which is not existing in vanilla js, but can be simulated... aka, the constructor of the Abstract throws an error if the class is instantiated, and therefore only allow its subclasses to be instantiated....
I am not sure of the value of having such Abstract behaviour, but this has been implemented in a few places in this code base.

File: tests/idv_service/session/retrieve/configuration/session.configuration.response.spec.js (hunk 4 above)
Comment (laurent-yoti) [reply tests/idv_service/session/retrieve/configuration/session.configuration.response.spec.js:12]: ok

**Review by dhivya-pattabhiraman — APPROVED** [review]

**Review by liamd-yoti — APPROVED** [review]
👍

**Pull request merged**